sklearn/metrics/tests/test_common.py

```diff
@@ -2250,6 +2254,21 @@ def yield_metric_checker_combinations(metric_checkers=array_api_metric_checkers)
 @pytest.mark.parametrize("metric, check_func", yield_metric_checker_combinations())
 def test_array_api_compliance(metric, array_namespace, device, dtype_name, check_func):
     check_func(metric, array_namespace, device, dtype_name)
+    if (
+        getattr(metric, "__name__", None) == "median_absolute_error"
+        and array_namespace == "array_api_strict"
+    ):
+        try:
+            import array_api_strict
+        except ImportError:
+            pass
+        else:
+            if device == array_api_strict.Device("device1"):
+                # See https://github.com/data-apis/array-api-strict/issues/134
+                pytest.xfail(
+                    "`_weighted_percentile` is affected by array_api_strict bug when "
+                    "indexing with tuple of arrays on non-'CPU_DEVICE' devices."
+                )
```

Comment:
This is not ideal. We have a similar xfail in a `_weighted_percentile` test:

https://github.com/scikit-learn/scikit-learn/blob/398e8feb73f3e422c9d51d5863eb270709af4742/sklearn/utils/tests/test_stats.py#L186-L196

Note that as we add array API support for more regression metrics, we will need to add them to the xfail, as several others use `_weighted_percentile`.

@ev-br - you did mention there could be an array-api-strict release soon though?